### PR TITLE
Change add_context_info method in typos

### PR DESCRIPTION
**Pull request by irinakhismatullina**
In utils.add_context_info() save context parts as strings (space-separated-tokens) , not lists.
Example: instead of `["get", "token"]` use `"get token"`. 

Simplifies storing datasets in .csv files.

Based on #626.

### CODE REVIEW

**Comment by vmarkovtsev** [comment]
@irinakhismatullina Now we need to rebase here

File: lookout/style/typos/corrector.py [hunk 1]
```diff
@@ -136,7 +136,8 @@ def suggest(self, data: pandas.DataFrame, candidates: str = None,
         :param save_candidates_file: Path to file to save candidates to (.csv.xz).
         :param n_candidates: Number of most probable candidates to return.
         :param return_all: False to return suggestions only for corrected tokens.
-        :return: Dictionary {id : [[candidate, correctness_proba]]}, candidates are sorted \
+        :param save_candidates_file: Path to file to save candidates to (.pkl).
```
Comment (m09) [inline lookout/style/typos/corrector.py]: This param is duplicated.

File: lookout/style/typos/corrector.py [hunk 2]
```diff
@@ -136,7 +136,8 @@ def suggest(self, data: pandas.DataFrame, candidates: str = None,
...
+        :return: Dictionary `{id : [[candidate, correctness_proba]]}`, candidates are sorted \
```
Comment (m09) [inline lookout/style/typos/corrector.py]: `[(candidate, correctness_proba), …]` will better reflect the fact that each value is a list of tuples (and this should probably be updated everywhere here and in ranker)

File: lookout/style/typos/generation.py [hunk 3]
```diff
@@ -204,7 +204,7 @@ def _get_candidate_tokens(self, typo_info: TypoInfo) -> Set[str]:
             candidate_tokens.extend(typo_neighbors)
 
             if len(typo_info.before + typo_info.after) > 0:
-                context_neighbors = self._closest(self._compound_vec(typo_info.before +
+                context_neighbors = self._closest(self._compound_vec(typo_info.before + " " +
```
Comment (m09) [inline lookout/style/typos/generation.py]: We use python2 format strings: `"%s %s" % (typo_info.before, typo_info.after)`

File: lookout/style/typos/generation.py [hunk 4]
```diff
@@ -233,7 +233,7 @@ def _generate_features(self, typo_info: TypoInfo, dist: int, typo_vec: numpy.nda
         """
         before_vec = self._compound_vec(typo_info.before)
         after_vec = self._compound_vec(typo_info.after)
-        context_vec = self._compound_vec(typo_info.before + typo_info.after)
+        context_vec = self._compound_vec(typo_info.before + " " + typo_info.after)
```
Comment (m09) [inline lookout/style/typos/generation.py]: same

File: lookout/style/typos/generation.py [hunk 5]
```diff
@@ -274,7 +274,8 @@ def _freq_relation(self, first_token: str, second_token: str) -> float:
         return -numpy.log((1.0 * self._freq(first_token) + 1e-5) /
                           (1.0 * self._freq(second_token) + 1e-5))
 
-    def _compound_vec(self, split: List[str]) -> numpy.ndarray:
+    def _compound_vec(self, split: str) -> numpy.ndarray:
```
Comment (m09) [inline lookout/style/typos/generation.py]: `split` should be renamed to something else since now the string is not split.

**Pull request merged**